server/lib/api/users.js

```diff
@@ -96,3 +139,71 @@ async function resetPassword(ctx, next) {
     ctx.status = 204
   })
 }
+
+async function verifyEmail(ctx, next, userSockets) {
+  const { id } = ctx.params
+  const { code } = ctx.query
+  const { email } = ctx.request.body
+
+  if (!id || !code || !isValidEmail(email)) {
+    throw new httpErrors.BadRequest('Invalid parameters')
+  }
+
+  const emailVerified = await useEmailVerificationCode(id, email, code)
+  if (!emailVerified) {
+    throw new httpErrors.BadRequest('Email verification code is invalid')
+  }
+
+  ctx.status = 204
+
+  // If the user is not logged in, our job is done here. They will receive the updated data with
+  // `emailVerified` set to `true` next time they log in.
+  if (!ctx.session.userId) return
```

Comment:
This feels like a possible attack vector to me, perhaps you should enforce that you have to be logged in to verify an email?

Reply:
Hmm, I thought it was usual to allow email verification without being logged in. What exact attack vector do you see potentially happening here?